cli/src/ui/commands/Chat.tsx

```diff
@@ -216,6 +223,38 @@ const CliChat: FC<CliChatProps> = ({ sId: requestedSId, agentSearch, message })
     setIsSelectingNewAgent(true);
   }, []);
 
+  const handleApprovalRequest = useCallback(
+    async (event: AgentActionSpecificEvent): Promise<boolean> => {
+      if (event.type !== "tool_approve_execution") {
+        assert(event, "Unexpected event type for approval request");
```

Comment:
This line is strange to me. What is this supposed to assert ?

Reply:
This just should never happen, the TS compiler does not infer it.

Reply:
Did you mean to throw instead of assert?